diff --git a/src/region.cpp b/src/region.cpp
index e66401f..8874658 100644
--- a/src/region.cpp
+++ b/src/region.cpp
@@ -7,6 +7,7 @@
 #include <string>
 #include <vector>
 #include <stdlib.h>
+#include <cstdint>
 
 #include "error.h"
 #include "region.h"
diff --git a/src/region.h b/src/region.h
index 713e5a0..cd64c64 100644
--- a/src/region.h
+++ b/src/region.h
@@ -10,6 +10,7 @@
 #include <string>
 #include <vector>
 #include <stdlib.h>
+#include <cstdint>
 
 #include "error.h"
 
