diff --git a/src/bam_io.cpp b/src/bam_io.cpp
index b69b115..41e909d 100644
--- a/src/bam_io.cpp
+++ b/src/bam_io.cpp
@@ -64,17 +64,7 @@ void BamHeader::parse_read_groups(const char *text){
 }
 
 void BamCramReader::clear_cram_data_structures(){
-  if (iter_ != NULL && in_->is_cram){
-    // Change the region for the CRAM file to be a single position and iterate through all of the records
-    // We use this to deallocate the memory created during CRAM IO, as deallocation only
-    // appears to happen once the iterator reads all of the associated records for the region
-    bam1_t *b = bam_init1();
-    cram_range cur_range = in_->fp.cram->range;
-    cram_range new_range = {cur_range.refid, cur_range.start, cur_range.start+1};
-    in_->fp.cram->range  = new_range;
-    while (sam_itr_next(in_, iter_, b) >= 0){}
-    bam_destroy1(b);
-
+  if (iter_ != NULL){
     hts_itr_destroy(iter_);
     iter_ = NULL;
   }
@@ -94,15 +84,8 @@ BamCramReader::BamCramReader(const std::string& path, std::string fasta_path)
     if (fasta_path.empty())
       printErrorAndDie("Must specify a FASTA reference file path for CRAM file " + path);
 
-    // Open the FASTA reference file for the CRAM
-    char* fasta = new char[fasta_path.size()+1];
-    for (size_t i = 0; i < fasta_path.size(); ++i)
-      fasta[i] = fasta_path[i];
-    fasta[fasta_path.size()] = '\0';
-
-    if (cram_load_reference(in_->fp.cram, fasta) < 0)
+    if (hts_set_fai_filename(in_, fasta_path.c_str()) < 0)
       printErrorAndDie("Failed to open FASTA reference file for CRAM file");
-    delete [] fasta;
   }
 
   // Read the header
@@ -119,9 +102,6 @@ BamCramReader::BamCramReader(const std::string& path, std::string fasta_path)
   iter_            = NULL;
   start_           = -1;
   end_             = -1;
-  min_offset_      = 0;
-  reuse_first_aln_ = false;
-  cram_done_       = false;
 }
 
 BamCramReader::~BamCramReader(){
@@ -139,74 +119,34 @@ BamCramReader::~BamCramReader(){
 }
 
 bool BamCramReader::SetChromosome(const std::string& chrom){
+  if (iter_ != NULL){
+    hts_itr_destroy(iter_);
+    iter_ = NULL;
+  }
+
   iter_            = sam_itr_querys(idx_, hdr_, chrom.c_str());
   chrom_           = chrom;
-  min_offset_      = 0;
-  reuse_first_aln_ = false;
 
   if (iter_ != NULL){
     start_     = 0;
     end_       = INT32_MAX;
-    cram_done_ = false;
     return true;
   }
   else {
     start_     = -1;
     end_       = -1;
-    cram_done_ = true;
     return false;
   }
 }
 
 bool BamCramReader::SetRegion(const std::string& chrom, int32_t start, int32_t end){
-  if (in_->is_cram && iter_ != NULL && chrom.compare(chrom_) == 0 && start >= start_){
-    // Determine if we can reuse the CRAM iterator from the previous region
-    // and if so, modify the iterator accordingly
-    bool can_reuse = false;
-    if (min_offset_ != 0 && in_->fp.cram != NULL && in_->fp.cram->ctr != NULL){
-      cram_container *ctr = in_->fp.cram->ctr;
-      if (ctr->slice != NULL){
-	cram_slice *slice = ctr->slice;
-	if (slice->ref_start+1 < first_aln_.Position() && slice->ref_end > start){
-	  can_reuse     = true;
-	  // When the min_offset_ is 0, we set it to 1 as 0 is the unset value. We therefore undo that change here
-	  // If the offset really was 1, we'll simply consider 1 more alignment that won't overlap the region
-	  slice->curr_rec = (min_offset_ == 1 ? 0 : min_offset_);
-
-	  // Modify the iterator's start coordinate to reflect the current region
-	  cram_range cur_range = in_->fp.cram->range;
-	  cram_range new_range = {cur_range.refid, start+1, cur_range.end};
-	  in_->fp.cram->range  = new_range;
-	}
-      }
-    }
-
-    if (can_reuse){
-      chrom_           = chrom;
-      start_           = start;
-      end_             = end;
-      cram_done_       = false;
-      min_offset_      = 0;
-      reuse_first_aln_ = false;
-      return true;
-    }
-
-    // If we can't reuse the CRAM iterator, we'll set the region as in all other cases by proceeding to the rest of the function
-  }
-
   if (iter_ != NULL){
-    clear_cram_data_structures();
     hts_itr_destroy(iter_);
     iter_ = NULL;
   }
 
-  // For CRAM files, as we try to reuse the iterator for future regions, we cannot set an end coordinate
-  // Otherwise, the CRAM container will be freed after reading the last alignment in the current region
   std::stringstream region;
-  if (in_->is_cram)
-    region << chrom << ":" << start+1;
-  else
-    region << chrom << ":" << start+1 << "-" << end;
+  region << chrom << ":" << start+1 << "-" << end;
 
   std::string region_str = region.str();
   iter_ = sam_itr_querys(idx_, hdr_, region_str.c_str());
@@ -214,59 +154,27 @@ bool BamCramReader::SetRegion(const std::string& chrom, int32_t start, int32_t e
     chrom_     = chrom;
     start_     = start;
     end_       = end;
-    cram_done_ = false;
-
-    bool reuse_offset = (!in_->is_cram && min_offset_ != 0 && chrom.compare(chrom_) == 0 && start >= start_);
-    if (reuse_offset)
-      if (iter_->n_off == 1 && min_offset_ >= iter_->off[0].u && min_offset_ <= iter_->off[0].v)
-	iter_->off[0].u = min_offset_;
-
-    if (reuse_offset && first_aln_.GetEndPosition() > start && first_aln_.Position() < end){
-      // NOTE: min_offset_ remains unchanged, as the first valid alignment is the current first alignment
-      reuse_first_aln_ = true;
-    }
-    else {
-      min_offset_      = 0;
-      reuse_first_aln_ = false;
-    }
     return true;
   }
   else {
     chrom_           = "";
     start_           = -1;
     end_             = -1;
-    min_offset_      = 0;
-    reuse_first_aln_ = false;
-    cram_done_       = true;
     return false;
   }
 }
 
 bool BamCramReader::GetNextAlignment(BamAlignment& aln){
   if (iter_ == NULL) return false;
-  if (cram_done_)    return false;
-
-  if (reuse_first_aln_){
-    reuse_first_aln_ = false;
-    aln = first_aln_;
-    return true;
-  }
 
   int ret = sam_itr_next(in_, iter_, aln.b_);
   if ((ret < 0) || aln.b_->core.pos > end_+1){
     if (ret < -1)
       printErrorAndDie("Invalid record encountered in " + path_ + ". Please ensure the BAM/CRAM is not truncated and is properly formatted");
 
-    if (in_->is_cram){
-      // Don't destroy the iterator, as we may want to reuse it later
-      cram_done_ = true;
-      return false;
-    }
-    else {
-      hts_itr_destroy(iter_);
-      iter_ = NULL;
-      return false;
-    }
+    hts_itr_destroy(iter_);
+    iter_ = NULL;
+    return false;
   }
 
   // Set up alignment instance variables
@@ -278,22 +186,6 @@ bool BamCramReader::GetNextAlignment(BamAlignment& aln){
   aln.pos_      = aln.b_->core.pos;
   aln.end_pos_  = bam_endpos(aln.b_);
 
-  if (min_offset_ == 0){
-    if (in_->is_cram){
-      assert(in_->fp.cram != NULL && in_->fp.cram->ctr != NULL);
-      cram_container *ctr = in_->fp.cram->ctr;
-      assert(ctr->slice != NULL and ctr->slice->curr_rec > 0);
-      first_aln_  = aln;
-      min_offset_ = ctr->slice->curr_rec - 1;
-      if (min_offset_ == 0)
-	min_offset_++;
-    }
-    else {
-      first_aln_  = aln;
-      min_offset_ = iter_->curr_off;
-    }
-  }
-
   return true;
 }
 
diff --git a/src/bam_io.h b/src/bam_io.h
index e0d47f7..febabc4 100644
--- a/src/bam_io.h
+++ b/src/bam_io.h
@@ -10,9 +10,10 @@
 #include <sstream>
 #include <vector>
 #include <sys/stat.h>
+#include <unistd.h>
 
 #include "htslib/bgzf.h"
-#include "htslib/cram/cram.h"
+#include "htslib/cram.h"
 #include "htslib/sam.h"
 
 #include "error.h"
@@ -451,17 +452,12 @@ private:
   std::string path_;
   BamHeader*  header_;
   bool shared_header_;
-  bool cram_done_;
 
   // Instance variables for the most recently set region
   hts_itr_t *iter_;        // Iterator
   std::string chrom_;      // Chromosome
   int32_t     start_;      // Start position
   int32_t     end_;        // End position
-  uint64_t    min_offset_; // Offset after first alignment. For BAMs, this is a memory offset, while for CRAMs its
-                           // the index of the first alignment in the CRAM slice
-  BamAlignment first_aln_; // First alignment
-  bool reuse_first_aln_;
 
   // Private unimplemented copy constructor and assignment operator to prevent operations
   BamCramReader(const BamCramReader& other);
diff --git a/src/bgzf_streams.h b/src/bgzf_streams.h
index 72e6d91..7a0799b 100644
--- a/src/bgzf_streams.h
+++ b/src/bgzf_streams.h
@@ -30,7 +30,7 @@ TODO:
 #include <ostream>
 #include <stdexcept>
 
-#include "htslib/htslib/bgzf.h"
+#include "htslib/bgzf.h"
 
 class bgzf_streambuf : public std::streambuf {
  private:
diff --git a/src/check_phasing.cpp b/src/check_phasing.cpp
index 12e6a76..fb899f8 100644
--- a/src/check_phasing.cpp
+++ b/src/check_phasing.cpp
@@ -1,6 +1,7 @@
 #include <getopt.h>
 #include <stdlib.h>
 #include <time.h>
+#include <unistd.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/src/denovos/denovo_main.cpp b/src/denovos/denovo_main.cpp
index 72fbfc8..2af3d41 100644
--- a/src/denovos/denovo_main.cpp
+++ b/src/denovos/denovo_main.cpp
@@ -1,6 +1,7 @@
 #include <getopt.h>
 #include <stdlib.h>
 #include <time.h>
+#include <unistd.h>
 
 #include <fstream>
 #include <iostream>
diff --git a/src/fasta_reader.h b/src/fasta_reader.h
index df34f3c..67f8860 100644
--- a/src/fasta_reader.h
+++ b/src/fasta_reader.h
@@ -14,7 +14,7 @@
 #include "stringops.h"
 
 extern "C" {
-#include "htslib/htslib/faidx.h"
+#include "htslib/faidx.h"
 }
 
 /*
diff --git a/src/seq_stutter_genotyper.cpp b/src/seq_stutter_genotyper.cpp
index 29e5766..4c1a7ae 100644
--- a/src/seq_stutter_genotyper.cpp
+++ b/src/seq_stutter_genotyper.cpp
@@ -27,7 +27,7 @@
 #include "SeqAlignment/RepeatBlock.h"
 
 #include "cephes/cephes.h"
-#include "htslib/htslib/kfunc.h"
+#include "htslib/kfunc.h"
 
 int max_index(double* vals, unsigned int num_vals){
   int best_index = 0;
diff --git a/src/vcf_reader.h b/src/vcf_reader.h
index 986266d..1994435 100644
--- a/src/vcf_reader.h
+++ b/src/vcf_reader.h
@@ -8,9 +8,9 @@
 #include <vector>
 
 extern "C" {
-#include "htslib/htslib/bgzf.h" 
-#include "htslib/htslib/tbx.h" 
-#include "htslib/htslib/vcf.h" 
+#include "htslib/bgzf.h"
+#include "htslib/tbx.h"
+#include "htslib/vcf.h"
 }
 
 #include "error.h"
-- 
2.52.0

